In my case, it's this error.

```log
error: builder for '/nix/store/i5h58awgi0ns0bmpg5zfhzk3bk6vld2c-armagetronad-dedicated-0.2.9.2.3.drv' failed with exit code 2;
       last 25 log lines:
       > g++ -DHAVE_CONFIG_H -I. -I..   -iquote . -iquote ..  -I/nix/store/b23a6vc2nkq9v1vs1g003djakr78h8j3-libxml2-2.13.2-dev/include/libxml2   -iquote ./tools  -iquote ./render -O2 -std=c++11 -Wno-unused-but-set-variable -Wno-unused-private-field -Wno-unused-function -Wno-long-long -c -o render/librender_a-rRender.o `test -f 'render/rRender.cpp' || echo './'`render/rRender.cpp
       > network/nKrawallPrivate.cpp: In static member function 'static int nKrawall::FetchURL(const tString&, const char*, std::ostream&, int)':
       > network/nKrawallPrivate.cpp:461:19: error: 'xmlNanoHTTPOpen' was not declared in this scope
       >   461 |     void * ctxt = xmlNanoHTTPOpen( fullURL.str().c_str(), NULL);
       >       |                   ^~~~~~~~~~~~~~~
       > network/nKrawallPrivate.cpp:467:14: error: 'xmlNanoHTTPReturnCode' was not declared in this scope
       >   467 |     int rc = xmlNanoHTTPReturnCode(ctxt);
       >       |              ^~~~~~~~~~~~~~~~~~~~~
       > network/nKrawallPrivate.cpp:478:15: error: 'xmlNanoHTTPRead' was not declared in this scope
       >   478 |         len = xmlNanoHTTPRead( ctxt, &buf, max );
       >       |               ^~~~~~~~~~~~~~~
       > network/nKrawallPrivate.cpp:483:5: error: 'xmlNanoHTTPClose' was not declared in this scope
       >   483 |     xmlNanoHTTPClose(ctxt);
       >       |     ^~~~~~~~~~~~~~~~
```

--- a/nixos/tests/armagetronad.nix
+++ b/nixos/tests/armagetronad.nix
@@ -13,7 +13,7 @@ let
 
     { imports = [ ./common/user-account.nix ./common/x11.nix ];
       hardware.graphics.enable = true;
-      virtualisation.memorySize = 256;
+      virtualisation.memorySize = 384;
       environment = {
         systemPackages = [ pkgs.armagetronad ];
         variables.XAUTHORITY = "/home/${user}/.Xauthority";

--- a/pkgs/games/armagetronad/default.nix
+++ b/pkgs/games/armagetronad/default.nix
@@ -23,6 +23,7 @@
 , protobuf
 , xvfb-run
 , gnugrep
+, nixosTests
 , dedicatedServer ? false
 }:
 
@@ -191,6 +192,7 @@ let
           {
             # Allow both a "dedicated" passthru and a passthru for all the options other than the latest version, which this is.
             dedicated = mkArmagetron fn true;
+            tests.armagetronad = nixosTests.armagetronad;
           };
 
       meta = with lib; {

--- a/pkgs/games/armagetronad/default.nix
+++ b/pkgs/games/armagetronad/default.nix
@@ -22,6 +22,7 @@
 , libxml2
 , protobuf
 , xvfb-run
+, gnugrep
 , dedicatedServer ? false
 }:
 
@@ -54,22 +55,21 @@ let
     # https://gitlab.com/armagetronad/armagetronad/-/commits/trunk/?ref_type=heads
     ${unstableVersionMajor} =
       let
-        rev = "e7f41fd26363e7c6a72f0c673470ed06ab54ae08";
-        hash = "sha256-Uxxk6L7WPxKYQ4CNxWwEtvbZjK8BqYNTuwwdleZ44Ro=";
+        rev = "391a74625c1222dd180f069f1b61c3e069a3ba8c";
+        hash = "sha256-fUY0dBj85k0QhnAoDzyBmmKmRh9oCYC6r6X4ukt7/L0=";
       in dedicatedServer: {
         version = "${unstableVersionMajor}-${builtins.substring 0 8 rev}";
         src = fetchArmagetron rev hash;
         extraBuildInputs = [ protobuf boost ]
           ++ lib.optionals (!dedicatedServer) [ glew ftgl freetype SDL2 SDL2_image SDL2_mixer ];
         extraNativeBuildInputs = [ bison ];
-        extraNativeInstallCheckInputs = lib.optionals (!dedicatedServer) [ xvfb-run ];
       };
 
     # https://gitlab.com/armagetronad/armagetronad/-/commits/hack-0.2.8-sty+ct+ap/?ref_type=heads
     "${latestVersionMajor}-sty+ct+ap" =
       let
-        rev = "a5bffe9dda2b43d330433f76f14eb374701f326a";
-        hash = "sha256-cNABxfg3MSmbxU/R78QyPOMwXGqJEamaFOPNw5yhDGE=";
+        rev = "5a17cc9fb6e1e27a358711afbd745ae54d4a8c60";
+        hash = "sha256-111C1j/hSaASGcvYy3//TyHs4Z+3fuiOvCmtcWLdFd4=";
       in dedicatedServer: {
         version = "${latestVersionMajor}-sty+ct+ap-${builtins.substring 0 8 rev}";
         src = fetchArmagetron rev hash;
@@ -136,7 +136,9 @@ let
       nativeBuildInputs = [ autoconf automake gnum4 pkg-config which python3 ]
         ++ (resolvedParams.extraNativeBuildInputs or []);
 
-      nativeInstallCheckInputs = resolvedParams.extraNativeInstallCheckInputs or [];
+      nativeInstallCheckInputs = [ gnugrep ]
+        ++ lib.optional (!dedicatedServer) xvfb-run
+        ++ (resolvedParams.extraNativeInstallCheckInputs or []);
 
       postInstall = lib.optionalString (!dedicatedServer) ''
         mkdir -p $out/share/{applications,icons/hicolor}
@@ -154,18 +156,21 @@ let
         else
           run="$bin"
         fi
+        echo "Checking game info:" >&2
         version="$($run --version || true)"
+        echo "  - Version: $version" >&2
         prefix="$($run --prefix || true)"
-        rubber="$($run --doc | grep -m1 CYCLE_RUBBER)"
-
-        echo "Version: $version" >&2
-        echo "Prefix: $prefix" >&2
-        echo "Docstring: $rubber" >&2
+        echo "  - Prefix: $prefix" >&2
+        rubber="$(($run --doc || true) | grep -m1 CYCLE_RUBBER)"
+        echo "  - Docstring: $rubber" >&2
 
         if [[ "$version" != *"${resolvedParams.version}"* ]] || \
            [ "$prefix" != "$out" ] || \
            [[ ! "$rubber" =~ ^CYCLE_RUBBER[[:space:]]+Niceness[[:space:]]factor ]]; then
+          echo "Something didn't match. :-(" >&2
           exit 1
+        else
+          echo "Everything is ok." >&2
         fi
       '';
 

